url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/lunar/ProblemProblem.java

```diff
@@ -0,0 +1,218 @@
+package lunar;
+
+import com.sun.xml.internal.bind.v2.model.runtime.RuntimeNonElementRef;
+import common.Params;
+import common.math.Vector2d;
+import controller.ShipController;
+import entity.spaceship.Spaceship;
+import main.MainView;
+import main.ParamEnums;
+import main.StatsVisualiser;
+import strategy.CMAHandler;
+import strategy.IStrategy;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class ProblemProblem implements IProblem<double[]> {
+
+    public int runIndex;
+    public ParamEnums param; // which parameter is being examined?
+
+    public List<LunarTerrain> lts;
+    public double comparisonValue;
+
+    public double valueRange = 10;
+    public double thrustLimit = 20;
+    public double lunarGravity = 10;
+    public double friction = 1.0;
+    public int randomSeed = 1000;
+    public int numLandingPads = 1;
+    public int landingPadSize = 5;
+    public double survivableVelocity = 15;
+    public double startingFuel = 10000;
+    public double proximityWeight = 1.0;
+    public double velocityWeight = 1.0;
+    public double fuelWeight = 0.0;
+    public double angleWeight = 0.0;
+
+    public ProblemProblem(int runIndex, ParamEnums param) {
+        this.runIndex = runIndex;
+        this.param = param;
+        lts = new ArrayList<LunarTerrain>();
+    }
+
+    public int nDim() {
+        return 13;
+    }
+
+    public double fitness(double[] x) {
+        setUpValues(x);
+
+        double score = 0;
+
+        // calculate values based on 1000 random samples
+        LunarProblem lp = new LunarProblem();
+        double randMeanFitness = 0;
+        double randStdDevFitness = 0;
+        double M2 = 0;
+        for(int j = 0; j<1000; j++) {
+            double fitness =  lp.fitness(getRandomIndividual(lp.nDim()));
+            double delta = fitness - randMeanFitness;
+            randMeanFitness = randMeanFitness + (delta/(j+1));
+            M2 = M2 + delta*(fitness - randMeanFitness);
+        }
+        randStdDevFitness = Math.sqrt(M2/1000);
+
+        // if a problem has an init std dev of 0 it is likely intractable and not worth solving
+        if(randStdDevFitness == 0) {
+            score = Double.POSITIVE_INFINITY;
+        } else {
+            // get evolved solutions
+            for(int i=0; i<3; i++) {
+                lp = new LunarProblem();
+                CMAHandler cma = new CMAHandler(lp, 0, 0, Params.valueRange, Params.popSize);
+
+                while((!cma.hasCompleted())) {
+                    cma.run();
+                }
+                cma.finish();
+
+                double endBestFitness = cma.getBestRecentFitness();
+                double subScore = ((endBestFitness - randMeanFitness)/randStdDevFitness);
+
+                score += subScore;
+            }
+            score /= 3;
+        }
+
+
+
+        return score;
+    }
+
+    private void setUpValues(double[] x) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.